metals/src/main/scala/scala/meta/internal/metals/codeactions/AddingBracesCodeAction.scala

```diff
@@ -0,0 +1,641 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Try
+
+import scala.meta.Defn
+import scala.meta.Pkg
+import scala.meta.Template
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta._
+import scala.meta.inputs.Position
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.lsp4j.CodeActionParams
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.{lsp4j => l}
+
+class AddingBracesCodeAction(
+    trees: Trees,
+    buffers: Buffers
+) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: CodeActionParams, token: CancelToken)(implicit
+      ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val textDocumentIdentifier = params.getTextDocument
+    val path = textDocumentIdentifier.getUri.toAbsolutePath
+    val cursorRange = params.getRange
+    val cursorPosition = cursorRange.getStart()
+    val maybeTree =
+      if (cursorRange.getStart == cursorRange.getEnd)
+        trees
+          .findLastEnclosingAt[Tree](
```

Comment:
```suggestion
          .findLastEnclosingAt[Block](
```
and then invoke:
```
createCodeActionForPotentialBlockHolder(
              tree.parent,
              path,
              document,
              tree,
              "val definition"
            ).toSeq
```

Reply:
I’m afraid that would exclude `Template`

Reply:
We can do that next in the same way.

Reply:
Or we can add a new method that will try and match the first out of two

Reply:
Something like:

```
  def findLastEnclosingAt[T1 <: Tree: ClassTag, T2 <: Tree: ClassTag](
      source: AbsolutePath,
      lspPos: l.Position
  ): Option[Tree] = {

    def loop(t: Tree, pos: Position): Option[Tree] = {
      t match {
        case _: T1 | _: T2=>
          enclosedChildren(t.children, pos)
            .flatMap(loop(_, pos))
            .orElse(Some(t))
        case other =>
          enclosedChildren(other.children, pos).flatMap(loop(_, pos))
      }
    }
    get(source).flatMap { tree =>
      val pos = lspPos.toMeta(tree.pos.input)
      loop(tree, pos)
    }

  }
```

Reply:
oh, ooops!
Just noticed that we are showing code actions on not just blocks themselves but also the BlockHolder trees which are not blocks. I mean, val, var, defn, ...